nearcore/src/state_sync.rs

```diff
@@ -965,11 +963,15 @@ impl StateDumper {
     // Checks which parts have already been uploaded possibly by other nodes
     // We use &mut so the do_state_sync_dump() future will be Send, which it won't be if we use a normal
     // reference because of the Chain field
-    async fn check_stored_parts(&mut self) {
+    fn check_stored_parts(&self) -> impl Future<Output = ()> {
         let CurrentDump::InProgress(dump) = &self.current_dump else {
-            return;
+            return futures::future::Either::Left(futures::future::ready(()));
```

Comment:
Would returning `Option<impl Future<...>>` instead be an (ahem) *option*?

Reply:
That's a trade-off between complexity on implementation side and caller side. I feel this is more convenient to use, because the returned future can be simply `await`ed instead of explicitly checking if it's not `None`.

Reply:
I guess [`MaybeFuture`](https://docs.rs/maybe-future/1.0.1/maybe_future/struct.MaybeFuture.html) would be a great fit for this, but pulling in a new dependency for a single usage is an overkill.